scripts/build.ts

```diff
@@ -116,119 +118,119 @@ async function treeviewIconFont() {
-			}
-			return [id, ...toArray(proto.alias)].map((name) => ({ name, title: rawTitles.get(id) ?? title }));
-		}))).flat();
-		data.push({ name: 'none', title: 'Plain text' });
-
-		/**
-		 * Tries to guess the name of a language given its id.
-		 *
-		 * @param name The language id.
-		 */
-		function guessTitle(name: string) {
-			return (name.substring(0, 1).toUpperCase() + name.substring(1)).replace(/s(?=cript)/, 'S');
-		}
-
-		return Object.fromEntries(
-			data.filter(({ name, title }) => guessTitle(name) !== title).map(({ name, title }) => [name, title])
-		);
-	}
-};
-
-const dataInsertPlugin: Plugin = {
-	name: 'data-insert',
-	async renderChunk(code, chunk) {
-		const pattern = /\/\*\s*(\w+)\[\s*\*\/[\s\S]*?\/\*\s*\]\s*\*\//g;
-
-		// search for placeholders
-		const contained = new Set<string>();
-		let m;
-		while ((m = pattern.exec(code))) {
-			contained.add(m[1]);
-		}
-
-		if (contained.size === 0) {
-			return null;
-		}
-
-		// fetch placeholder data
-		const dataByName: Record<string, unknown> = {};
-		for (const name of contained) {
-			if (name in dataToInsert) {
-				dataByName[name] = await dataToInsert[name as keyof typeof dataToInsert]();
-			} else {
-				throw new Error(`Unknown placeholder ${name} in ${chunk.fileName}`);
-			}
-		}
-
-		// replace placeholders
-		const str = new MagicString(code);
-		str.replace(pattern, (_, name: string) => {
-			return JSON.stringify(dataByName[name]);
-		});
-		return toRenderedChunk(str);
-	},
-};
-
-const inlineRegexSourcePlugin: Plugin = {
-	name: 'inline-regex-source',
-	renderChunk(code) {
-		const str = new MagicString(code);
-		str.replace(
-			/\/((?:[^\n\r[\\\/]|\\.|\[(?:[^\n\r\\\]]|\\.)*\])+)\/\s*\.\s*source\b/g,
-			(m, source: string) => {
-				// escape backslashes
-				source = source.replace(/\\(.)|\[(?:\\s\\S|\\S\\s)\]/g, (m, g1: string) => {
-					if (g1) {
-						// characters like /\n/ can just be kept as "\n" instead of being escaped to "\\n"
-						if (/[nrt0/]/.test(g1)) {
-							return m;
-						}
-						if ('\\' === g1) {
-							return '\\\\\\\\'; // escape using 4 backslashes
-						}
-						return '\\\\' + g1;
-					} else {
-						return '[^]';
-					}
-				});
-				// escape single quotes
-				source = source.replace(/'/g, "\\'");
-				// wrap source in single quotes
-				return "'" + source + "'";
-			}
-		);
-		return toRenderedChunk(str);
-	},
-};
+// const dataToInsert = {
+// 	aliases_placeholder: async () => {
+// 		const data = await Promise.all([...languageIds, ...pluginIds].map(async (id) => {
+// 			const proto = await loadComponent(id);
+// 			return { id, alias: toArray(proto.alias) };
+// 		}));
+// 		return Object.fromEntries(data.flatMap(({ id, alias }) => alias.map((a) => [a, id])));
+// 	},
+// 	all_languages_placeholder: () => Promise.resolve(languageIds),
+// 	title_placeholder: async () => {
+// 		const rawTitles = new Map<string, string>();
+// 		for (const [id, entry] of Object.entries(components.languages)) {
```

Comment:
- [ ] todo: figure if/how to handle components.json. For now I simply disabled `buildJS`.